examples/build_file_generation/BUILD

```diff
@@ -54,3 +61,22 @@ py_binary(
     visibility = ["//:__subpackages__"],
     deps = [":build_file_generation"],
 )
+
+# Generates an entrypoint for pytest that is picked up by Gazelle for the py_test target below.
+py_pytest_main(
+    name = "__test__",
+)
```

Comment:
How come `py_pytest_main` becomes a dependency to a `py_test`, instead of subsuming the `py_test` rule? 

Wholly replacing the `py_test` rule with a macro/rule/rule+macro seems a cleaner user interface.